example/factories.ts

```diff
@@ -30,14 +30,19 @@ export const fileDownloadEndpointsFactory = new EndpointsFactory(createResultHan
 export const fileStreamingEndpointsFactory = new EndpointsFactory(createResultHandler({
   getPositiveResponse: () => createApiResponse(z.file().binary(), 'image/*'),
   getNegativeResponse: () => createApiResponse(z.string(), getType('txt') || 'text/plain'),
-  handler: ({response, error, output}) => {
+  handler: async ({response, error, output}) => {
     if (error) {
       response.status(400).send(error.message);
       return;
     }
     if ('filename' in output) {
-      fs.createReadStream(output.filename)
-        .pipe(response.type(output.filename));
+      // @todo update readme and put in the changelog
+      await new Promise((resolve, reject) =>
+        fs.createReadStream(output.filename)
+          .pipe(response.type(output.filename))
+          .on('close', resolve)
+          .on('error', reject)
+      );
```

Comment:
I don't like this requirement, though last resort handler does it job in place.
Perhaps I'll return to this solution in the future.